app/controllers/return_to_sp_controller.rb

```diff
@@ -1,28 +1,24 @@
-class ReturnToSpController < ApplicationController
+class ReturnToSpController < Redirect::RedirectController
   before_action :validate_sp_exists
 
   def cancel
     redirect_url = sp_return_url_resolver.return_to_sp_url
     analytics.track_event(Analytics::RETURN_TO_SP_CANCEL, redirect_url: redirect_url)
-    redirect_to redirect_url
+    redirect_to_and_log redirect_url
   end
 
   def failure_to_proof
     redirect_url = sp_return_url_resolver.failure_to_proof_url
     analytics.track_event(
       Analytics::RETURN_TO_SP_FAILURE_TO_PROOF,
       redirect_url: redirect_url,
-      **idv_location_params,
+      **location_params,
     )
-    redirect_to redirect_url
+    redirect_to_and_log redirect_url
```

Comment:
Probably shouldn't want to double log here:

* Keep `RETURN_TO_SP_FAILURE_TO_PROOF` and do a plain redirect
* Remove `RETURN_TO_SP_FAILURE_TO_PROOF` and somehow embed "failure to proof" as an extra logging detail in `redirect_to_and_log` (e.g. keyword arguments)

Reply:
>Probably shouldn't want to double log here. Instead, one of:
>
>* Keep `RETURN_TO_SP_FAILURE_TO_PROOF` and do a plain redirect
>* Remove `RETURN_TO_SP_FAILURE_TO_PROOF` and somehow embed "failure to proof" as an extra logging detail in `redirect_to_and_log` (e.g. keyword arguments)

Went with a third option in ee9feb0: Keep `redirect_to_and_log`, but allow overriding the event name, so as to log as `RETURN_TO_SP_FAILURE_TO_PROOF`.